src/datasets/vtr_dataset.py

```diff
@@ -50,8 +63,17 @@ def collate_function(self, batch: list[tuple[torch.Tensor, int]]) -> dict[str, t
         for i, s in enumerate(slices):
             attention_mask[i, : len(s)] = 1
 
-        return {
-            "slices": batched_slices,
-            "attention_mask": attention_mask,
-            "labels": torch.tensor(labels, dtype=torch.float),
-        }
+        if self.ocr_flag:
+            texts = [item[2] for item in batch]
+            return {
+                "slices": batched_slices,
+                "attention_mask": attention_mask,
+                "labels": torch.tensor(labels, dtype=torch.float),
+                "texts": texts,
+            }
+        else:
+            return {
+                "slices": batched_slices,
+                "attention_mask": attention_mask,
+                "labels": torch.tensor(labels, dtype=torch.float),
+            }
```

Comment:
```python
batch = {
    "slices": batched_slices,
     "attention_mask": attention_mask,
     "labels": torch.tensor(labels, dtype=torch.float),
}
if self.ocr_flag:
     batch["texts"] = [item[2] for item in batch]
return batch
```

Reply:
For now, you are returning texts as strings. At what moment do you replace it with target ids? Usually, it happens in the Dataset class, look `BERTDataset` for example.